src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -409,17 +411,14 @@ public void RegisterForCancellation(CancellationToken cancellationToken)
                         var @this = (GetAddrInfoExState)o!;
                         int cancelResult = 0;
 
-                        lock (@this)
+                        GetAddrInfoExContext* context = @this.CancellationContext;
```

Comment:
Because the lock was removed, a volatile read is needed here to ensure visibility.

Reply:
This is outdated now after a2a9355.